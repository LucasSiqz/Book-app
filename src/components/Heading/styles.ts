import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 1.5rem;
`

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.darkGray};
    font-size: ${theme.font.sizes.large};
  `}
`

export const Nav = styled.nav`
  ${({ theme }) => css`
    a {
      font-size: ${theme.font.sizes.small};
      color: ${theme.colors.blue};
      text-decoration: none;

      &:hover {
        text-decoration: underline;
      }
    }
  `}
`
