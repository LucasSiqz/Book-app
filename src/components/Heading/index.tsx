import Link from 'next/link'

import * as S from './styles'

export type HeadingProps = {
  title: string
  nav: {
    title: string
    path: string
  }
}

const Heading = ({
  title = 'Discover new book',
  nav = { title: 'More', path: '/books' }
}: HeadingProps) => (
  <S.Wrapper>
    <S.Title>{title}</S.Title>
    <S.Nav aria-labelledby="footer resources">
      <Link href={`${nav.path}`}>
        <a>{nav.title}</a>
      </Link>
    </S.Nav>
  </S.Wrapper>
)

export default Heading
