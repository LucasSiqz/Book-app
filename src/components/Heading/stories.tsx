import { Story, Meta } from '@storybook/react/types-6-0'

import Heading, { HeadingProps } from '.'

export default {
  title: 'Heading',
  component: Heading,
  args: {
    title: 'Discover new book',
    nav: {
      title: 'More',
      path: '/books'
    }
  }
} as Meta

export const Default: Story<HeadingProps> = (args) => <Heading {...args} />
