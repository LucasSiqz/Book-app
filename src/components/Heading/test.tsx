import { render, screen } from 'utils/test-utils'

import Heading from '.'

const props = {
  title: 'Discover new book',
  nav: {
    title: 'More',
    path: '/books'
  }
}

describe('<Heading />', () => {
  it('should render correctly', async () => {
    const { container } = render(<Heading {...props} />)

    expect(screen.getByRole('link', { name: /More/i })).toHaveAttribute(
      'href',
      '/books'
    )
    expect(await screen.findByText(/Discover new book/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
