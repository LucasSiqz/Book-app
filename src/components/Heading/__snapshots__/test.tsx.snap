// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 1.5rem;
}

.c1 {
  color: #3F4043;
  font-size: 1.8rem;
}

.c2 a {
  font-size: 1.4rem;
  color: #4ABDF1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    Discover new book
  </h2>
  <nav
    aria-labelledby="section navigation"
    class="c2"
  >
    <a
      href="/books"
    >
      More
    </a>
  </nav>
</div>
`
